src/components/media-player/plyr/youtube-player.js

```diff
@@ -0,0 +1,318 @@
+        : youtubeThumbnailUrl.lowResThumbnailUrl;
+    }
+  }, [posterImageUrl, youtubeThumbnailUrl]);
+
+  const triggerPlay = useCallback(() => {
+    if (player && !!sourceDurationInMs) {
+      setShowPosterImage(false);
+
+      if (playbackEnded) {
+        player.currentTime = startTimeInS;
+        setPlaybackEnded(false);
+      }
+
+      player.play();
+    }
+  }, [player, startTimeInS, playbackEnded, sourceDurationInMs]);
+
+  const triggerPause = useCallback(() => {
+    player?.pause();
+  }, [player]);
+
+  const triggerSeek = useCallback(seekedTimeWithinRangeInMs => {
+    if (player) {
+      const startTimeInMs = startTimeInS * 1000;
+      const currentActualTimeInMs = startTimeInMs + seekedTimeWithinRangeInMs;
+      player.currentTime = currentActualTimeInMs / 1000;
+
+      if (player.paused) {
+        const currentTimeWithinRangeInMs = currentActualTimeInMs - startTimeInMs;
+        onProgress(currentTimeWithinRangeInMs);
+      }
+      setPlaybackEnded(false);
+    }
+  }, [player, startTimeInS, onProgress]);
+
+  const setProgressInterval = callback => {
+    clearInterval(progressInterval.current);
+    progressInterval.current = null;
+    if (callback) {
+      progressInterval.current = setInterval(callback, MEDIA_PROGRESS_INTERVAL_IN_MILLISECONDS);
+    }
+  };
+
+  useEffect(() => {
+    if (!sourceDurationInMs) {
+      return;
+    }
+    const options =  {
+      controls: [],
+      ratio: aspectRatio,
+      clickToPlay: true,
+      loadSprite: false,
+      blankVideo: '',
+      fullscreen: { enabled: false, fallback: false },
+      // https://developers.google.com/youtube/player_parameters#Parameters
+      youtube: {
+        autoplay: 0,
+        rel: 0,
+        fs: 0,
+        showinfo: 0,
+        disablekb: 1,
+        iv_load_policy: 3,
+        modestbranding: 1,
+        controls: 0
+      }
+    };
+
+    if (startTimeInS) {
+      options.youtube.start = startTimeInS;
+    }
+
+    if (endTimeInS) {
+      options.youtube.end = endTimeInS;
+    }
+
+    const playerInstance = new Plyr(plyrRef.current, options);
+    setPlayer(playerInstance);
+  }, [plyrRef, aspectRatio, sourceDurationInMs, startTimeInS, endTimeInS]);
+
+  useEffect(() => {
+    if (player) {
+      player.source = {
+        type: audioOnly ? 'audio': 'video',
+        sources: [{ src: sourceUrl,  provider: 'youtube' }]
+      };
+    }
+  }, [player, sourceUrl, audioOnly]);
+
+  useEffect(() => {
+    if (player) {
+      player.speed = playbackRate;
+    }
+  }, [player, playbackRate]);
+
+  useEffect(() => {
+    if (player) {
+      player.volume = volume;
+    }
+  }, [player, volume]);
+
+  const handleEnded = useCallback(() => {
+    onEnded();
+    setProgressInterval(null);
+    setPlaybackEnded(true);
+    // compensate for cases where youtube actual source duration is shorter than reported,
+    // thus having playback end 1s earlier; likely a sounding issue on their side
```

Comment:
Or even likelier a **r**ounding issue (although it has sound)

Reply:
Such are the wa[ys]/[ves] of the brain 😂 Corrected, thanks!